src/components/Modals/components/Modal.js

```diff
@@ -32,12 +32,11 @@ export const Modal = ({
   const okButton = useRef(null);
 
   useEffect(() => {
-    okButton.current?.focus();
+    const button = okButton.current;
+    button.focus();
 
-    // the ref must be saved into a variable to accomodate react warning
-    const elementToClean = okButton.current;
     return () => {
```

Comment:
i am probably asking this question for the 100th time now. why do we need to return a function here?

Reply:
😄 no worries. This is the standard syntax React expects to run a clean up function with [useEffect()](https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1). Basically the callback function runs when the component un-mounts. With class components it was simply done with [componentWillUnmount()](https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1)

Reply:
I thought that in order to avoid memory leakages we need to clean up **any** event listeners when un-mounting, but after researching a bit it does not look like it is the case with `focus()` and `blur()` because we are not doing something `onFocus()` or `onBlur()`, we are just focusing on something. My confusion! Thanks for pointing that out.

Secondly, I found out that there is a `autoFocus` attribute we can pass to an element to do exactly the same. So why not using it!! I've added that

Reply:
❤️

Reply:
👍